### ISSUE TITLE
Multiple issues in f5os_snmp

### ISSUE THREAD

**Issue by MVogel91** [F5Networks/f5-ansible-f5os#12]
##### COMPONENT NAME
f5os_snmp

##### ANSIBLE VERSION

```
# ansible --version
ansible [core 2.17.1]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.10/dist-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.4
  libyaml = True
```
##### F5OS VERSION
```
1.5.2-21056
```
##### OS / ENVIRONMENT

N/A

##### SUMMARY

1. v2c communities are not idempotent
2. Targets can't be configured independently of communities in separate tasks

##### STEPS TO REPRODUCE

1. Create SNMP v2c Community, rerun playbook

```
  tasks:
    - name: Create SNMP communities
      f5networks.f5os.f5os_snmp:
        snmp_community:
          - name: "{{ item.community.name }}"
            security_model: "{{ item.community.versions }}"
        snmp_target:
          - name: "{{ item.name }}"
            security_model: "{{ item.version }}"
            ipv4_address: "{{ item.ipv4_address }}"
            port: "{{ item.port }}"
            community: "{{ item.community.name }}"
      with_items: "{{ snmp.targets }}"
```

2. Have 2 separate tasks for community and target
```
  tasks:
    - name: Create SNMP communities
      f5os_snmp:
        snmp_community:
          - name: "{{ item.name }}"
            security_model: "{{ item.version }}"
      with_items: "{{ snmp.communities }}"
      when: snmp is defined and snmp.communities is defined

    - name: Create SNMP targets
      f5os_snmp:
        snmp_target:
          - name: "{{ item.name }}"
            security_model: "{{ item.version }}"
            ipv4_address: "{{ item.ipv4_address }}"
            port: "{{ item.port }}"
            community: "{{ item.community }}"
      with_items: "{{ snmp.targets }}"
      when: snmp is defined and snmp.targets is defined
```

##### EXPECTED RESULTS

1. Repeatedly pasting the same declaration for v2c communities results to ok/unchanged.
2. Target hosts can be configured independently of communities

##### ACTUAL RESULTS

1. Module fails with "TypeError: string indices must be integers"
```
The full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-local-5213vva558b8/ansible-tmp-1719304746.3832297-5310-280915669646745/AnsiballZ_f5os_snmp.py", line 107, in <module>
    _ansiballz_main()
  File "/root/.ansible/tmp/ansible-local-5213vva558b8/ansible-tmp-1719304746.3832297-5310-280915669646745/AnsiballZ_f5os_snmp.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/root/.ansible/tmp/ansible-local-5213vva558b8/ansible-tmp-1719304746.3832297-5310-280915669646745/AnsiballZ_f5os_snmp.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.f5networks.f5os.plugins.modules.f5os_snmp', init_globals=dict(_module_fqn='ansible_collections.f5networks.f5os.plugins.modules.f5os_snmp', _modlib_path=modlib_path),
  File "/usr/lib/python3.10/runpy.py", line 224, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 565, in <module>
  File "/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 558, in main
  File "/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 350, in exec_module
  File "/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 364, in present
  File "/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 385, in update
  File "/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 461, in update_on_device
  File "/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/module_utils/common.py", line 141, in api_params
  File "/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 228, in snmp_community
TypeError: string indices must be integers
failed: [10.170.9.37] (item={'name': 'target1', 'version': 'v2c', 'ipv4_address': '192.0.2.5', 'port': 1234, 'community': {'name': 'community1', 'versions': ['v2c']}}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": {
        "community": {
            "name": "community1",
            "versions": [
                "v2c"
            ]
        },
        "ipv4_address": "192.0.2.5",
        "name": "target1",
        "port": 1234,
        "version": "v2c"
    },
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-local-5213vva558b8/ansible-tmp-1719304746.3832297-5310-280915669646745/AnsiballZ_f5os_snmp.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-local-5213vva558b8/ansible-tmp-1719304746.3832297-5310-280915669646745/AnsiballZ_f5os_snmp.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-local-5213vva558b8/ansible-tmp-1719304746.3832297-5310-280915669646745/AnsiballZ_f5os_snmp.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.f5networks.f5os.plugins.modules.f5os_snmp', init_globals=dict(_module_fqn='ansible_collections.f5networks.f5os.plugins.modules.f5os_snmp', _modlib_path=modlib_path),\n  File \"/usr/lib/python3.10/runpy.py\", line 224, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.10/runpy.py\", line 96, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.10/runpy.py\", line 86, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 565, in <module>\n  File \"/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 558, in main\n  File \"/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 350, in exec_module\n  File \"/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 364, in present\n  File \"/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 385, in update\n  File \"/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 461, in update_on_device\n  File \"/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/module_utils/common.py\", line 141, in api_params\n  File \"/tmp/ansible_f5networks.f5os.f5os_snmp_payload_u3i55mmt/ansible_f5networks.f5os.f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 228, in snmp_community\nTypeError: string indices must be integers\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

2. Module fails for "TypeError: 'NoneType' object is not iterable"
```
The full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-local-5090167lsjl6/ansible-tmp-1719304476.314065-5188-166467802791312/AnsiballZ_f5os_snmp.py", line 107, in <module>
    _ansiballz_main()
  File "/root/.ansible/tmp/ansible-local-5090167lsjl6/ansible-tmp-1719304476.314065-5188-166467802791312/AnsiballZ_f5os_snmp.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/root/.ansible/tmp/ansible-local-5090167lsjl6/ansible-tmp-1719304476.314065-5188-166467802791312/AnsiballZ_f5os_snmp.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.f5networks.f5os.plugins.modules.f5os_snmp', init_globals=dict(_module_fqn='ansible_collections.f5networks.f5os.plugins.modules.f5os_snmp', _modlib_path=modlib_path),
  File "/usr/lib/python3.10/runpy.py", line 224, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_f5os_snmp_payload_785p57gn/ansible_f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 565, in <module>
  File "/tmp/ansible_f5os_snmp_payload_785p57gn/ansible_f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 558, in main
  File "/tmp/ansible_f5os_snmp_payload_785p57gn/ansible_f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 350, in exec_module
  File "/tmp/ansible_f5os_snmp_payload_785p57gn/ansible_f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 363, in present
  File "/tmp/ansible_f5os_snmp_payload_785p57gn/ansible_f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py", line 404, in exists
TypeError: 'NoneType' object is not iterable
failed: [10.170.9.37] (item={'name': 'target1', 'version': 'v1', 'ipv4_address': '192.0.2.5', 'port': 1234, 'community': {'name': 'community1', 'versions': ['v1']}}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": {
        "community": {
            "name": "community1",
            "versions": [
                "v1"
            ]
        },
        "ipv4_address": "192.0.2.5",
        "name": "target1",
        "port": 1234,
        "version": "v1"
    },
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-local-5090167lsjl6/ansible-tmp-1719304476.314065-5188-166467802791312/AnsiballZ_f5os_snmp.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-local-5090167lsjl6/ansible-tmp-1719304476.314065-5188-166467802791312/AnsiballZ_f5os_snmp.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-local-5090167lsjl6/ansible-tmp-1719304476.314065-5188-166467802791312/AnsiballZ_f5os_snmp.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.f5networks.f5os.plugins.modules.f5os_snmp', init_globals=dict(_module_fqn='ansible_collections.f5networks.f5os.plugins.modules.f5os_snmp', _modlib_path=modlib_path),\n  File \"/usr/lib/python3.10/runpy.py\", line 224, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.10/runpy.py\", line 96, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.10/runpy.py\", line 86, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_f5os_snmp_payload_785p57gn/ansible_f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 565, in <module>\n  File \"/tmp/ansible_f5os_snmp_payload_785p57gn/ansible_f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 558, in main\n  File \"/tmp/ansible_f5os_snmp_payload_785p57gn/ansible_f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 350, in exec_module\n  File \"/tmp/ansible_f5os_snmp_payload_785p57gn/ansible_f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 363, in present\n  File \"/tmp/ansible_f5os_snmp_payload_785p57gn/ansible_f5os_snmp_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_snmp.py\", line 404, in exists\nTypeError: 'NoneType' object is not iterable\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

**Comment by pgouband** [comment]
Hi,

Thanks for reporting. Added to the backlog and internal tracking ID for this request is: INFRAANO-1606.

**Comment by RavinderReddyF5** [comment]
Hi @MVogel91 ,
fixed in v1.10.1 collection, please verify and reopen bug if issue seen
